modules/opc-publisher/docker/linux/amd64/Dockerfile

```diff
@@ -1,23 +1,27 @@
-ARG runtime_base_tag=2.2-runtime-alpine
-ARG build_base_tag=2.2-sdk-alpine
+FROM mcr.microsoft.com/dotnet/core/runtime:2.2-alpine AS base
+FROM mcr.microsoft.com/dotnet/core/sdk:2.2 AS build
+ENV PROJECT_ROOT=modules/opc-publisher
 
-FROM microsoft/dotnet:${build_base_tag} AS build
+ARG BUILD_SOURCEVERSION
+ARG RELEASE_BUILD
+ENV PROJECT_BUILD=1
 WORKDIR /app
+COPY *.props /
+WORKDIR /app/project
+COPY NuGet.Config NuGet.Config
+COPY ${PROJECT_ROOT}/*.props /
+COPY ${PROJECT_ROOT}/opcpublisher/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.csproj opcpublisher/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.csproj
+ENV BUILD_SOURCEVERSION=${BUILD_SOURCEVERSION}
+ENV RELEASE_BUILD=${RELEASE_BUILD}
+RUN dotnet restore --configfile NuGet.Config -nowarn:msb3202,nu1503 opcpublisher/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.csproj
 
-# copy csproj and restore as distinct layers
-COPY opcpublisher/*.csproj ./opcpublisher/
-WORKDIR /app/opcpublisher
-RUN dotnet restore
+FROM build AS publish
+COPY ${PROJECT_ROOT}/opcpublisher ./opcpublisher
+WORKDIR /app/project/opcpublisher
+RUN dotnet publish "Microsoft.Azure.IIoT.Modules.OpcUa.Publisher.csproj" -c Release -o /out
 
-# copy and publish app
+FROM base AS final
 WORKDIR /app
-COPY opcpublisher/. ./opcpublisher/
-WORKDIR /app/opcpublisher
-RUN dotnet publish -c Release -o out
-
-# start it up
-FROM microsoft/dotnet:${runtime_base_tag} AS runtime
-WORKDIR /app
-COPY --from=build /app/opcpublisher/out ./
+COPY --from=publish /out .
 WORKDIR /appdata
-ENTRYPOINT ["dotnet", "/app/opcpublisher.dll"]
\ No newline at end of file
+ENTRYPOINT ["dotnet", "/app/opcpublisher.dll"]
```

Comment:
we should rename the publisher project file to something like Microsoft.Azure.IIoT.Modules.OpcUa.Publisher for consistency

Reply:
The project file is already named as such.  Just the output file is not.  I leave this until the overall publisher changes that are coming soon.